### DISCUSSION TITLE
Really really need a way of identifying custom nodes.

### DISCUSSION THREAD

**Discussion by Ferniclestix** [comfyanonymous/ComfyUI#1161] [Ideas]
Heya, I've been making tons of tutorials for comfyUI recently and find myself with the problem of being unable to identify whether some nodes are custom nodes or what custom pack they are from. It makes it very difficult to do tutorials if I can't say, oh this is from blah pack. and it would also help with bug reporting, I bet plenty of people report problems with custom nodes that cannot be answered on the comfyUI issues tab because they don't know where the nodes are from either.

![Capturesuggest](https://github.com/comfyanonymous/ComfyUI/assets/128430314/38bdf059-2f43-4c2d-a42a-9f832ee4c38c)
Something like this would be cool where you can click to open the nodes github page.

**Comment by ltdrdata** [comment]
Oh that is a nice idea.
One issue is that if a short nickname or something similar is not provided for the custom node, it will become quite ugly.
The specifications of the custom node need to be slightly expanded.

**Reply by Ferniclestix** [reply]
wouldn't be as bad if some custom nodes didn't end up in the default comfyUI folders :<

**Comment by ntdviet** [comment]
I had the issue too so I ended up adding a "Packname/...." to the nodes myself lol

**Comment by donengel72** [comment]
In general, having the node locations in the node properties would be really nice. IE, /vextranodes/hue rotation

**Comment by ltdrdata** [comment]
https://www.youtube.com/watch?v=c73mcPLtPTI
![nickname](https://github.com/comfyanonymous/ComfyUI/assets/128333288/8685d921-6f0b-4789-8cf5-27dbe5d001a9)

Now this is supported in ComfyUI-Manager.

To node developers:
If you want to display a nickname other than the title I assigned, please write a docstring inside the .py file. Nicknames exceeding 20 characters will be truncated.

"""
@author: Dr.Lt.Data
@title: Impact Pack
@nickname: Impact Pack
@description: This extension offers various detector nodes and detailer nodes that allow you to configure a workflow that automatically enhances facial details. And provide iterative upscaler.
"""

**Reply by Ferniclestix** [reply]
Uh, love it, testing it and there is some odd behaviour around minimizing and resizing.  other than that, seems to work good :D

![Capture](https://github.com/comfyanonymous/ComfyUI/assets/128430314/d2934752-8fd1-4c1c-bb99-f3ba7b84bdcf)

**Reply by Ruffy314** [reply]
Maybe add a very small hash to the display, in case two packs accidentally (or maliciously) try to use the same nick name? Not sure if that's something we have to worry about.

**Reply by ltdrdata** [reply]
> Uh, love it, testing it and there is some odd behaviour around minimizing and resizing.  other than that, seems to work good :D
> 
> ![Capture](https://github.com/comfyanonymous/ComfyUI/assets/128430314/d2934752-8fd1-4c1c-bb99-f3ba7b84bdcf)
> 

oh.. I need to fix that... lol
